app/assets/javascripts/discourse/app/mixins/user-fields-validation.js

```diff
@@ -20,27 +20,42 @@ export default Mixin.create({
     this.set("userFields", userFields);
   },
 
-  // Validate required fields
   @discourseComputed("userFields.@each.value")
   userFieldsValidation() {
-    let userFields = this.userFields;
-    if (userFields) {
-      userFields = userFields.filterBy("field.required");
+    if (!this.userFields) {
+      return EmberObject.create({ ok: true });
     }
-    if (!isEmpty(userFields)) {
-      const emptyUserField = userFields.find((uf) => {
-        const val = uf.get("value");
-        return !val || isEmpty(val);
-      });
-      if (emptyUserField) {
-        const userField = emptyUserField.field;
-        return EmberObject.create({
+
+    this.userFields.forEach((userField) => {
+      let validation = EmberObject.create({ ok: true });
+
+      if (userField.field.required && isEmpty(userField.value)) {
+        validation = EmberObject.create({
+          failed: true,
+          reason: I18n.t("user_fields.required", {
+            name: userField.field.name,
+          }),
+          element: userField.field.element,
+        });
+      } else if (
+        this.accountPassword &&
+        userField.value === this.accountPassword
```

Comment:
Can we do a `include` check rather than an `equality` check? That way we're more user friendly (towards whitespaces for example)

```suggestion
        userField.value.include(this.accountPassword)
```

Reply:
We could even go as far as adding `.toLowerCase()` to both values to make the `include` less case sensitive.

Reply:
I would also like it to do it the other way too, `this.accountPassword.include(userField.value)`. What do you think?